openslides_backend/action/actions/option/update.py

```diff
@@ -126,16 +126,11 @@ def _handle_poll_option_data(
         self, instance: Dict[str, Any], poll: Dict[str, Any]
     ) -> None:
         data = self._get_data(instance)
-
         if poll.get("type") == "analog":
-            if poll.get("pollmethod") == "N":
-                instance["no"] = data.get("no", "0.000000")
-            else:
-                instance["yes"] = data.get("yes", "0.000000")
-                if poll.get("pollmethod") in ("YN", "YNA"):
-                    instance["no"] = data.get("no", "0.000000")
-                if poll.get("pollmethod") == "YNA":
-                    instance["abstain"] = data.get("abstain", "0.000000")
+            pollmethod = poll["pollmethod"]
+            for key in ("yes", "no", "abstain"):
+                if key[0].upper() in pollmethod:
```

Comment:
Maybe better to set all not given keys to -2 not_documented without `if`

Reply:
My preferred solution would be to just prevent keys which do not match the pollmethod - the client should not even display them and there should be no input for them anyway. Changing the values even thought the pollmethod does not match feels just as wrong as ignoring it.

Reply:
IMO the keys not matching the pollmethod should be `None` (best solution), or -2, not documented, but 0.0000 is a valid, setted voting value. Maybe this is just your test setup?